src/pages/ProfilePage/index.tsx

```diff
@@ -0,0 +1,190 @@
+import { useContext, useState, useEffect } from "react";
+import { AuthContext } from "../../AuthContext";
+import { Button } from "../../components/Button/index";
+import {
+  addDoc,
+  collection,
+  deleteDoc,
+  doc,
+  getDocs,
+  updateDoc,
+} from "firebase/firestore";
+import { db } from "../../firebase";
+import { QuoteCard } from "../../components/QuoteCard";
+import { useQuotesContext } from "../../QuotesContextProvider";
+
+export const ProfilePage = () => {
+  const authContext = useContext(AuthContext);
+  const [quoteInput, setQuoteInput] = useState<string>("");
+  const [authorInput, setAuthorInput] = useState<string>("");
+  const [message, setMessage] = useState<string | null>(null);
+
+  const [editQuoteId, setEditQuoteId] = useState<string | null>(null);
+  const [editQuoteInput, setEditQuoteInput] = useState("");
+  const [editAuthorInput, setEditAuthorInput] = useState("");
+  const [addNewQuote, setaddNewQuote] = useState<boolean>(false);
+
+  const quotes = useQuotesContext();
+  const userEmail = authContext?.user?.email?.toUpperCase().split("@")[0];
+  const userID = authContext?.user?.uid;
+  const isMessagePositive = message?.includes("unsuccessfully") == false;
+
+  useEffect(() => {
+    if (message) {
+      const timeout = setTimeout(() => {
+        setMessage(null);
+      }, 3000);
+
+      return () => clearTimeout(timeout);
+    }
+  }, [message]);
+
+  async function addQuote(e: React.FormEvent) {
```

Comment:
Make sure that you add validation for authorInput and quoteInput fields. What it users submits empty form? What if users submits form with random values instead of proper strings? What if users submits a malicious code instead of authorInput and/or quoteInput?

Make sure to add some feedback in the form that will guide the user to fill the form correctly.